### ISSUE TITLE
feat: Add article about payments

### ISSUE THREAD

**Issue by KnorpelSenf** [grammyjs/website#155]
Add an article about payments.

**Comment by LWJerri** [comment]
@KnorpelSenf, that's all good, but I'd like to know what exactly do we need to write? Any steps on how to set up payment or anything like that? It would be nice to create a small base of questions or topics that need to be covered.

**Comment by KnorpelSenf** [comment]
My typical approach to writing these pages is:
1. What does this feature do? (Bot Dev accept payments from users, link to TG docs)
2. How do payments on TG work? (TG doesn't handle the money itself, so how is this set up? Payment providers? What's the flow of data? What's the flow of money?)
3. What do I need to get started? (Credit card? Any accounts? Any tokens/secrets?)
4. How can this be coded with grammY? (Example code snippets for each relevant step in 2.)
5. Useful tips, advanced settings, links to other resources, how to move on from here, anything else?

**Comment by LWJerri** [comment]
Okay, then I will prepare some information base and try to write something.

**Comment by KnorpelSenf** [comment]
Feel free to share early drafts so we can iterate on the changes together :)

**Comment by LWJerri** [comment]
> Feel free to share early drafts so we can iterate on the changes together :)

Hi. I have some references on how to do these docs, but currently, study year is an outing and I have exams, etc.

**Comment by LWJerri** [comment]
Soooo, this is my structure for the Telegram payments guide. @KnorpelSenf, what do you think about it?

1. Introduction
- Provide a brief overview of the purpose of the Payments Guide.
- Explain the importance of understanding payment integration for bot development on the platform.
2. Feature Overview
- Describe the functionality of the payment feature in bot development, highlighting its benefits and potential use cases.
- Include a link to the official Telegram documentation for accepting payments from users.
3. Understanding Payments on Telegram
- Explain how payments on TG work, emphasizing that TG doesn't handle the money itself.
- Outline the general setup process and the involvement of payment providers.
- Describe the flow of data between TG, payment providers, and the bot.
- Explain the flow of money from users to the bot developer's account.
4. Prerequisites
- List the requirements and prerequisites for getting started with accepting payments on TG.
- Mention if a credit card is needed, any specific accounts that need to be set up, or any tokens/secrets required for integration.
5. Integration Steps with grammY
- Provide step-by-step instructions on how to code the payment integration using the grammY framework.
- Include example code snippets for each relevant step, demonstrating how to set up and handle payments within a bot.
6. Advanced Topics and Tips
- Offer additional tips, tricks, and best practices for working with payments on TG.
- Include information on advanced settings and customization options for payment integration.
- Provide links to other relevant resources, such as official TG documentation or community forums for further exploration.
7. Conclusion and Next Steps
- Summarize the key points covered in the Payments Guide.
- Encourage users to explore the provided information and experiment with payment integration.
- Suggest possible next steps or related topics for further learning.
8. Review and Feedback
- Invite users to provide feedback on the documentation, ensuring clarity and completeness.
- Consider involving other individuals or stakeholders to review and offer their opinions on the guide.

**Comment by KnorpelSenf** [comment]
> 1. Introduction
> 
> * Provide a brief overview of the purpose of the Payments Guide.

There is no need to say why a guide a necessary. When people open a guide for payments, they already have a reason to read it. What you can do is to include an introducing sentence about the purpose of payments themselves.

> * Explain the importance of understanding payment integration for bot development on the platform.

Yeap, but don't drag this out. Again, users who read this have a reason already.

> 2. Feature Overview
> 
> * Describe the functionality of the payment feature in bot development, highlighting its benefits and potential use cases.
> * Include a link to the official Telegram documentation for accepting payments from users.

Great. Also consider including limitations. It's good to have a differentiated view, not advertise the feature too much (just a little).

> 3. Understanding Payments on Telegram
> 
> * Explain how payments on Telegram work, emphasizing that Telegram doesn't handle the money itself.
> * Outline the general setup process and the involvement of payment providers.
> * Describe the flow of data between Telegram, payment providers, and the bot.
> * Explain the flow of money from users to the bot developer's account.
> 
> 4. Prerequisites
> 
> * List the requirements and prerequisites for getting started with accepting payments on Telegram.
> * Mention if a credit card is needed, any specific accounts that need to be set up, or any tokens/secrets required for integration.

Awesome

> 5. Integration Steps with grammY
> 
> * Provide step-by-step instructions on how to code the payment integration using the grammY framework.
> * Include example code snippets for each relevant step, demonstrating how to set up and handle payments within a bot.

Nice. I usually expect readers to stop reading here because their bot works now.

> 6. Advanced Topics and Tips
> 
> * Offer additional tips, tricks, and best practices for working with payments on Telegram.
> * Include information on advanced settings and customization options for payment integration.
> * Provide links to other relevant resources, such as official Telegram documentation or community forums for further exploration.

Great to include this, I'm curious what you'll link here.

> 7. Conclusion and Next Steps
> 
> * Summarize the key points covered in the Payments Guide.
> * Encourage users to explore the provided information and experiment with payment integration.
> * Suggest possible next steps or related topics for further learning.

Drop the summary. If we feel like we need a summary then the page isn't concise enough. Next steps are cool, but they can also be included along the way (“if you want to know more about this step, look here”) rather than at the end of the page.

> 8. Review and Feedback
> 
> * Invite users to provide feedback on the documentation, ensuring clarity and completeness.
> * Consider involving other individuals or stakeholders to review and offer their opinions on the guide.

This sounds like ChatGPT generated it. Sounds pointless to me. It isn't related to payments in any way and we already have this content written down elsewhere. What are the stakeholders even?

**Comment by LWJerri** [comment]
> > 1. Introduction
> > 
> > 
> > * Provide a brief overview of the purpose of the Payments Guide.
> 
> There is no need to say why a guide a necessary. When people open a guide for payments, they already have a reason to read it. What you can do is to include an introducing sentence about the purpose of payments themselves.
> 
> > * Explain the importance of understanding payment integration for bot development on the platform.
> 
> Yeap, but don't drag this out. Again, users who read this have a reason already.
> 
> > 2. Feature Overview
> > 
> > 
> > * Describe the functionality of the payment feature in bot development, highlighting its benefits and potential use cases.
> > * Include a link to the official Telegram documentation for accepting payments from users.
> 
> Great. Also consider including limitations. It's good to have a differentiated view, not advertise the feature too much (just a little).
> 
> > 3. Understanding Payments on Telegram
> > 
> > 
> > * Explain how payments on Telegram work, emphasizing that Telegram doesn't handle the money itself.
> > * Outline the general setup process and the involvement of payment providers.
> > * Describe the flow of data between Telegram, payment providers, and the bot.
> > * Explain the flow of money from users to the bot developer's account.
> > 
> > 
> > 4. Prerequisites
> > 
> > 
> > * List the requirements and prerequisites for getting started with accepting payments on Telegram.
> > * Mention if a credit card is needed, any specific accounts that need to be set up, or any tokens/secrets required for integration.
> 
> Awesome
> 
> > 5. Integration Steps with grammY
> > 
> > 
> > * Provide step-by-step instructions on how to code the payment integration using the grammY framework.
> > * Include example code snippets for each relevant step, demonstrating how to set up and handle payments within a bot.
> 
> Nice. I usually expect readers to stop reading here because their bot works now.
> 
> > 6. Advanced Topics and Tips
> > 
> > 
> > * Offer additional tips, tricks, and best practices for working with payments on Telegram.
> > * Include information on advanced settings and customization options for payment integration.
> > * Provide links to other relevant resources, such as official Telegram documentation or community forums for further exploration.
> 
> Great to include this, I'm curious what you'll link here.
> 
> > 7. Conclusion and Next Steps
> > 
> > 
> > * Summarize the key points covered in the Payments Guide.
> > * Encourage users to explore the provided information and experiment with payment integration.
> > * Suggest possible next steps or related topics for further learning.
> 
> Drop the summary. If we feel like we need a summary then the page isn't concise enough. Next steps are cool, but they can also be included along the way (“if you want to know more about this step, look here”) rather than at the end of the page.
> 
> > 8. Review and Feedback
> > 
> > 
> > * Invite users to provide feedback on the documentation, ensuring clarity and completeness.
> > * Consider involving other individuals or stakeholders to review and offer their opinions on the guide.
> 
> This sounds like ChatGPT generated it. Sounds pointless to me. It isn't related to payments in any way and we already have this content written down elsewhere. What are the stakeholders even?

I took all my writing from my notes, then my correspondence with my friends and threw it into Chat GPT, yes. That was about a week ago. Then I removed the unnecessary theses and the last item was added by me, we have such a thing in plugins, but since you don't think it's necessary, then okay.

**Comment by LWJerri** [comment]
I had to organize everything somehow, because there was a lot of different information, which was even duplicated. Chat GPT organized it for me and gave me a brief summary of what should have been done.

**Comment by KnorpelSenf** [comment]
> we have such a thing in plugins

What are you referring to?

**Comment by LWJerri** [comment]
> I took all my writing from my notes, then my correspondence with my friends and threw it into Chat GPT, yes. That was about a week ago. Then I removed the unnecessary theses and the last item was added by me, we have such a thing in plugins, but since you don't think it's necessary, then okay.

![image](https://github.com/grammyjs/website/assets/50290430/24946f78-30eb-4953-bff8-86344219ded6)

I'm say about this.

**Comment by KnorpelSenf** [comment]
Yes. This is one of the places where we provide information about how to contribute, but it's done once for all plugins. In the guide, we will not do this after every single article. Same for the advanced section or the docs of a particular plugin.
